packages/suite/src/utils/suite/parseUri.ts

```diff
@@ -38,4 +40,38 @@ const parseUri = (uri: string): ParsedURI => {
     };
 };
 
-export { parseUri };
+interface BaseProtocol {
+    scheme: string;
+    address: string;
+}
+
+interface BitcoinProtocol extends BaseProtocol {
+    scheme: PROTOCOL_SCHEME;
+    amount: number;
+}
+
+export const getProtocolInfo = (uri: string): BitcoinProtocol => {
+    const { protocol, pathname, search } = new URL(uri.replace('://', ':'));
+    const scheme = protocol.slice(0, -1);
+
+    const params: { [key: string]: string } = {};
+
+    new URLSearchParams(search).forEach((v, k) => {
+        params[k] = v;
+    });
+
+    if (
+        scheme === PROTOCOL_SCHEME.BITCOIN &&
+        typeof Number.parseFloat(params.amount) === 'number'
```

Comment:
this condition is not working, because `typeof NaN === 'number`

Reply:
```suggestion
        !Number.isNaN(Number.parseFloat(params.amount))
```

Reply:
9c2e9b22e0309afac357bd4773e0ca574efc61d5